### PR TITLE
feat: execution workflow diffUTR

### LINKED ISSUE #27: Execution Workflow: diffUTR
## WHAT    
**Write** execution workflow for [diffUTR](https://github.com/ETHZ-INS/diffUTR). Use the provided [small files for testing](https://github.com/iRNA-COSI/APAeval/tree/main/tests/test_data) (running the workflow on real data is a different issue).

## CHECKLIST   
* [ ] Use [snakemake template](https://github.com/iRNA-COSI/APAeval/tree/main/docs/templates/snakemake) or [nextflow template](https://github.com/iRNA-COSI/APAeval/tree/main/docs/templates/nextflow) to create your workflow.
* [ ] Comment your code
* [ ] Run individual rules/processes in either conda envs or docker/singularity containers for reproducibility
* [ ] Input:  .bam or .fastq from [test_data]((https://github.com/iRNA-COSI/APAeval/tree/main/tests/test_data))
* [ ] Give [feedback about the method](https://forms.gle/mAAkbRJqhedJnSeJ6)

## OUTPUTS (see [specification](https://github.com/iRNA-COSI/APAeval/blob/main/execution_workflows/execution_output_specification.md)): 
* [ ] ~Output: Adhere to output specification for Identification challenge~ No identification
* [ ] ~Output: Adhere to output specification for quantification challenge~ No quantification
* [ ] Output: Adhere to output specification for differential usage challenge
   > This TSV file contains two columns:

    > - gene ID
    > - significance of differential PAS usage

    > Column names should not be added to the file.

### PR DESCRIPTION
**Pull request by chilampoon**
Fixes #27 

## Checklist

- [x] I have performed a self-review of my own code
- [x] My code follows the templates/style guidelines of the repository
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my code

Figuring out how to dockerize

### CODE REVIEW

**Comment by FitzsimmonsCM** [comment]
@all-contributor add @chilampoon for code

**Comment by FitzsimmonsCM** [comment]
@all-contributors please add @chilampoon for code

**Review by uniqueg — DISMISSED** [review]
Looks good to me

File: execution_workflows/diffUTR/diffUTR_nextflow/nextflow.config [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  enabled = true
+  file = "${params.tracedir}/execution_trace.txt"
+}
+dag {
+  enabled = true
+  file = "${params.tracedir}/pipeline_dag.svg"
+}
+
+manifest {
+  name = 'APAeval'
```
Comment (dominikburri) [inline execution_workflows/diffUTR/diffUTR_nextflow/nextflow.config:91]: Replace *name*, *author*, *description* with the actual values. I guess @yuukiiwa can say what is expected here.

File: execution_workflows/diffUTR/diffUTR_nextflow/results/pipeline_info/execution_timeline.html [hunk 2]
```diff
@@ -0,0 +1,200 @@
+<!doctype html>
```
Comment (dominikburri) [inline execution_workflows/diffUTR/diffUTR_nextflow/results/pipeline_info/execution_timeline.html:1]: It is great that the `results` are shown to us and that the method works, but I don't think this should be in the repository. 
Pplease remove the `results` directory. We should be able to run `test_data` on this method, once it is merged.

**Review by dominikburri — CHANGES_REQUESTED** [review]
Looks good, but some clarifications needed. 
* Extend `README.md`
* Make sure that `input.csv` runs with the directory structure we have (that is, obtaining `test_data/` from appropriate relative path. 
* Remove `results/`.

**Comment by chilampoon** [comment]
@dominikburri the reason why I didn't point to files in `test_data/` is that they are just 1 control & 1 case, so I couldn't get the result in diffUTR since no replicates. 
But I think I can use `try()` to prevent stopping the execution and also print out the warning/error.

**Comment by dominikburri** [comment]
> @dominikburri the reason why I didn't point to files in `test_data/` is that they are just 1 control & 1 case, so I couldn't get the result in diffUTR since no replicates.
> But I think I can use `try()` to prevent stopping the execution and also print out the warning/error.

Thank you for the explanation, good to know that the method only works with at least two replicates (each?).
I don't think the `try()` clause is necessary right now, just make sure to mention the need for replicates in `README.md`.

File: execution_workflows/diffUTR/diffUTR_nextflow/results/pipeline_info/pipeline_dag.svg [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (ninsch3000) [inline execution_workflows/diffUTR/diffUTR_nextflow/results/pipeline_info/pipeline_dag.svg:1]: Can you copy the dag into the `diffUTR_nextflow` directory and include it in the `README`? We do that for the snakemake template and would also be nice here!    
Agree with Dominik that the rest of the results should not be included in the repo.

File: execution_workflows/diffUTR/diffUTR_nextflow/build_docker_image.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+## Build Docker image from this directory with enviroment.yml and Dockerfile
+```
+docker build . -t [execution_tool]:1.0
```
Comment (ninsch3000) [inline execution_workflows/diffUTR/diffUTR_nextflow/build_docker_image.md:3]: Can this be used inside `diffUTR_nextflow`? If yes, fill in correct names, if no, remove file from directory.

File: execution_workflows/diffUTR/diffUTR_nextflow/results/pipeline_info/pipeline_dag.svg (hunk 3 above)
Comment (chilampoon) [reply execution_workflows/diffUTR/diffUTR_nextflow/results/pipeline_info/pipeline_dag.svg:1]: Oops I just deleted it...

**Review by dominikburri — APPROVED** [review]

**Review by yuukiiwa — APPROVED** [review]
Looking good! The `environment.yml` was not used in the `Dockerfile`, so I will just delete that later.

**Pull request merged**